syntax="proto2";
package azino;

message TxOpStatus {
  enum Code {
    Ok = 0;
    Conflicts = 1;
    TooLate = 2;
    Block = 3;
    NotExist = 4;
    PartitionErr = 5;
  };
  optional Code error_code = 1 [default = Ok];
  optional string error_message = 2;
}

message TxStatus {
  enum Code {
   UnInitialize = 0;
   Start = 1;
   Preput = 2;
   Abort = 3;
   Commit = 4;
   Abnormal = 5;
  };
  optional Code status_code = 1 [default = UnInitialize];
  optional string status_message = 2;
}

message TxIdentifier {
  optional uint64 start_ts = 1;
  optional uint64 commit_ts = 2;
  optional TxStatus status = 3;
}